common/persistence/sql/sqlplugin/mysql/typeconv.go

```diff
@@ -22,30 +22,33 @@ package mysql
 
 import "time"
 
-var (
-	minMySQLDateTime = getMinMySQLDateTime()
-)
-
-type (
-	// DataConverter defines the API for conversions to/from
-	// go types to mysql datatypes
-	DataConverter interface {
-		ToMySQLDateTime(t time.Time) time.Time
-		FromMySQLDateTime(t time.Time) time.Time
-	}
-	converter struct{}
-)
+// DataConverter defines the API for conversions to/from
+// go types to mysql datatypes
+type DataConverter interface {
+	ToDateTime(t time.Time) time.Time
+	FromDateTime(t time.Time) time.Time
+}
+
+// Converter implements DataConverter
+type Converter struct{}
+
+// NewConverter returns a new instance of DataConverter
+func NewConverter() *Converter {
+	return &Converter{}
+}
+
+var minMySQLDateTime = getMinMySQLDateTime()
 
-// ToMySQLDateTime converts to time to MySQL datetime
-func (c *converter) ToMySQLDateTime(t time.Time) time.Time {
+// ToDateTime converts to time to MySQL datetime
+func (c *Converter) ToDateTime(t time.Time) time.Time {
```

Comment:
these seem very mysql-specific

Reply:
ah, nevermind.  these are part of an interface, so we can swap out implementations (mysql looks rather special), so that's not really an issue for code-reuse.

I'd probably make this a `mysqlconverter` in that case though.

Reply:
Decided to keep it private `converter` with mysql-specific logic and implement `sqlite.converter` with no-op methods. `DataConverter` was already exposed. Don't think that moving `DataConverter` to a specific package will be useful because it's already in `mysql` package.